src/model/transaction/RegisterNamespaceTransaction.ts

```diff
@@ -79,7 +79,9 @@ export class RegisterNamespaceTransaction extends Transaction {
             new UInt64([0, 0]),
             NamespaceType.SubNamespace,
             namespaceName,
-            new NamespaceId(subnamespaceNamespaceId(parentNamespace, namespaceName)),
+            typeof parentNamespace === 'string' ?
```

Comment:
you are using *type definition* as the way to determine whether you should be creating a subnamespace id or a namespace id ? I believe it would be best to to handle this with the namespace.type content.

Reply:
This is not to determine the type of the namespace, it determine the type of the input whether a string (name) or uint (number[]). if a number[] passed in, we don't need to call 'subnamespaceNamespaceId' to convert the string to number[].